### PR TITLE
#2246: fix confirmation modal fadeout animation

### LINKED ISSUE #2246: `ConfirmationModal` fadeout animation missing
It looks like the fadeout animation on ConfirmationModal is missing on Cancel and Confirm actions. One place this can be observed is in the Admin Console, [on the GroupDetail page's "Delete Group" action](http://github.com/pixiebrix/pixiebrix-app/blob/0b1fbb4f9e3cbf064d5656147b061c5054c422f3/pixiebrix-app/src/pages/groups/GroupDetailPage.tsx#L45-L50).

Related:
https://github.com/pixiebrix/pixiebrix-app/pull/904#discussion_r774251375

### PR DESCRIPTION
**Pull request by abdullah4244**
Fixes #2246 
https://www.loom.com/share/9e2137087dd3447893fd8ce290c60c35

### CODE REVIEW

**Comment by twschiller** [comment]
Thanks @abdullah4244, @mnholtz could you help review?

@abdullah4244, could you you please add the ConfirmationModal to Storybook? The story could include a button to toggle the modal (in the story you may need to defined a wrapper component that has a button and the useModals hook)

**Comment by abdullah4244** [comment]
> Thanks @abdullah4244, @mnholtz could you help review?
> 
> @abdullah4244, could you you please add the ConfirmationModal to Storybook? The story could include a button to toggle the modal (in the story you may need to defined a wrapper component that has a button and the useModals hook)

got it

**Comment by mnholtz** [comment]
@abdullah4244 thank you for your PR! The animation seems to be working smoothly. I'm noticing a flickering on the fade-out where the original confirm body is shown:

https://user-images.githubusercontent.com/36575242/169113103-db5d7500-d8a2-46f3-8f5b-d7958395a266.mov

Would it be possible to fix this?

**Comment by abdullah4244** [comment]
> @abdullah4244 thank you for your PR! The animation seems to be working smoothly. I'm noticing a flickering on the fade-out where the original confirm body is shown:
> 
>  Screen.Recording.2022-05-18.at.11.03.14.AM.mov 
> Would it be possible to fix this?


> @abdullah4244 thank you for your PR! The animation seems to be working smoothly. I'm noticing a flickering on the fade-out where the original confirm body is shown:
> 
>  Screen.Recording.2022-05-18.at.11.03.14.AM.mov 
> Would it be possible to fix this?

@mnholtz 
that flickering was basically happening because of setting ModalProps to null while it was transitioning out now i have updated the check can you review and check code

**Comment by twschiller** [comment]
@abdullah4244 it looks like your pre-commit hooks might not be set up properly? https://github.com/pixiebrix/pixiebrix-extension/wiki/Local-build-setup#requirements Reach out on the `#engineering` channel if you have questions on setup

File: src/components/ConfirmationModal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const Template: Story<StoryType>= (args) => {
+    const {title,message,submitCaption,cancelCaption} = args;
+    
+return (
+  <ModalProvider {...args} >
+      <ChildComponent title={title} message={message} submitCaption={submitCaption} cancelCaption={cancelCaption}/>
+  </ModalProvider>
+)};
+
+export const WithoutProps = Template.bind({});
```
[the remark this answers is not in the document]
Comment (abdullah4244) [reply src/components/ConfirmationModal.stories.tsx:69]: ok got it

[the remark this answers is not in the document]
Comment (mnholtz) [reply src/components/ConfirmationModal.stories.tsx:69]: Second what Todd said 👆

File: src/components/ConfirmationModal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+         // Do any action here if confirm === true
+       }, [showConfirmation]);
+       return (
+        <AsyncButton onClick={buttonAction}>
+        Confirm Modal
+        </AsyncButton>
+       )
+  }
+
+const Template: Story<StoryType>= (args) => {
```
Comment (mnholtz) [inline src/components/ConfirmationModal.stories.tsx:60]: Nit: I think using convenience types `ComponentStory` and `ComponentMeta` makes storybook stories more legible (and consistent as most of our stories are written this way).

**Review by mnholtz — APPROVED** [review]
LGTM. See minor code refactoring opportunities, but once pre-commit hooks are run this will be good to go.

File: src/components/ConfirmationModal.stories.tsx (hunk 2 above)
Comment (abdullah4244) [reply src/components/ConfirmationModal.stories.tsx:60]: @mnholtz i basically wanted to add extra controls in storybook for this component like adding title , description , button text of modal but ModalProvider components wasn't taking these keys as props that's why I created Custom Type and used Story here , and i got hint of this from one of the stories within a project

**Pull request merged**